source/adapters/native_cpu/enqueue.cpp

```diff
@@ -215,34 +214,32 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
       auto numGroups = groups.size();
       auto groupsPerThread = numGroups / numParallelThreads;
       auto remainder = numGroups % numParallelThreads;
-      for (unsigned thread = 0; thread < numParallelThreads; thread++) {
-        futures.emplace_back(
-            tp.schedule_task([groups, thread, groupsPerThread,
+      for (unsigned thread = 0; groupsPerThread && thread < numParallelThreads;
```

Comment:
I can see the logic behind `groupsPerThread &&` in avoiding running a loop where each loop body does nothing, but personally, I think it's clearer to have this as a `if (groupsPerThread) { ... }` around the loop.